src/main/java/org/broadinstitute/hellbender/tools/funcotator/FilterFuncotations.java

```diff
@@ -0,0 +1,201 @@
+        funcotationFilters.add(new LmmFilter());
+    }
+
+    @Override
+    public void apply(final VariantContext variant, final ReadsContext readsContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        outputVcfWriter.add(applyFilters(variant, getMatchingFilters(variant)));
+    }
+
+    /**
+     * Collect the names of the {@link FuncotationFilter}s matching the Funcotations of the given variant.
+     *
+     * The filter will be treated as a match if it matches Funcotations for any of the transcripts in the
+     * variant's Funcotation map.
+     */
+    private Set<String> getMatchingFilters(final VariantContext variant) {
+        final Set<String> matchingFilters = new HashSet<>();
+
+
+        final Map<Allele, FuncotationMap> funcs = FuncotatorUtils.createAlleleToFuncotationMapFromFuncotationVcfAttribute(
+                funcotationKeys, variant, "Gencode_" + referenceVersion.gencodeVersion + "_annotationTranscript", "FAKE_SOURCE");
+
+        funcs.values().forEach(funcotationMap -> {
+            final Stream<Map<String, String>> transcriptFuncotations = funcotationMap.getTranscriptList().stream()
+                    .map(funcotationMap::get)
+                    .map(funcotations -> funcotations.stream()
+                            .flatMap(this::extractFuncotationFields)
+                            .filter(entry -> entry.getValue() != null && !entry.getValue().isEmpty())
+                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
+            transcriptFuncotations.forEach(funcotations -> {
+                final Set<String> matches = funcotationFilters.stream()
+                        .filter(f -> f.checkFilter(funcotations))
+                        .map(FuncotationFilter::getFilterName)
+                        .collect(Collectors.toSet());
+                matchingFilters.addAll(matches);
+            });
+        });
+
+        return matchingFilters;
+    }
+
+    /**
+     * Parse the entries in a Funcotation into a stream of map entries.
+     */
+    private Stream<Map.Entry<String, String>> extractFuncotationFields(final Funcotation funcotation) {
+        return funcotation.getFieldNames().stream()
+                .map(name -> new AbstractMap.SimpleEntry<>(name, funcotation.getField(name)));
+    }
+
+    /**
+     * Mark a variant as matching a set of Funcotation filters, or as matching no filters.
+     */
+    private VariantContext applyFilters(final VariantContext variant, final Set<String> matchingFilters) {
+        final VariantContextBuilder variantContextBuilder = new VariantContextBuilder(variant);
+        final boolean isSignificant = !matchingFilters.isEmpty();
+
+        // Mark the individual filters that make the variant significant, if any.
+        final String clinicalSignificance =
+                isSignificant ? String.join(",", matchingFilters) : FilterFuncotationsConstants.CLINSIG_INFO_NOT_SIGNIFICANT;
```

Comment:
As mentioned in a comment above, can you extract the comma here into a constant in this class?